src/Shared/Constants.cs

```diff
@@ -97,6 +97,7 @@ internal static class MSBuildConstants
         internal static readonly char[] ForwardSlash = { '/' };
         internal static readonly char[] ForwardSlashBackslash = { '/', '\\' };
         internal static readonly char[] WildcardChars = { '*', '?' };
+        internal static readonly string[] CharactersForExpansion = { "*", "?", "$(", "@(", "%" };
```

Comment:
This feels off to me but I can't articulate why so it's probably fine.

Reply:
It would probably be more robust to explicitly (try to) expand every item spec fragment and see if it changed from the original, but at that point, I'm doing a nontrivial amount of the work for the whole operation, which felt undesirable from a perf perspective.